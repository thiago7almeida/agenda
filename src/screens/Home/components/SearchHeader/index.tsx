import React, {useCallback} from 'react';
import {useStyleSheet, Layout, Input, Icon} from '@ui-kitten/components';

import themedStyles from './themedStyles';

const SearchHeader: React.FC = (): React.ReactElement => {
  const styles = useStyleSheet(themedStyles);
  const [searchQuery, setSearchQuery] = React.useState<string>();
  const setValue = (newValue: string) => setSearchQuery(newValue);

  const accessoryRight = useCallback(
    () => <Icon style={styles.icon} fill="#8F9BB3" name="search" />,
    [],
  );

  return (
    <Layout style={styles.header} level="1">
      <Input
        placeholder="Buscar"
        value={searchQuery}
        onChangeText={setValue}
        accessoryRight={accessoryRight}
      />
    </Layout>
  );
};

export default SearchHeader;
