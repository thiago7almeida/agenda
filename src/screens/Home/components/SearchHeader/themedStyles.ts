import {StyleService} from '@ui-kitten/components';

export default StyleService.create({
  list: {
    flex: 1,
  },
  header: {
    paddingHorizontal: 16,
    paddingTop: 16,
    paddingBottom: 8,
  },
  icon: {
    width: 22,
    height: 22,
  },
  item: {
    paddingVertical: 16,
    borderBottomWidth: 1,
    borderBottomColor: 'background-basic-color-3',
  },
});
