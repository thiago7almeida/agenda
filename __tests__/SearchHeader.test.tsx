import React from 'react';

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchHeader from '../src/screens/Home/components/SearchHeader';
import * as redux from 'react-redux';

const spy = jest.spyOn(redux, 'useSelector');
spy.mockReturnValue({data: [], name: ''});

const spyOnUseDispatch = jest;
spyOnUseDispatch.spyOn(redux, 'useDispatch').mockReturnValue(jest.fn());

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useFocusEffect: () => jest.fn(),
    useNavigation: () => {
      navigation: () => jest.fn();
    },
  };
});

describe('SearchHeader', () => {
  describe('Rendering', () => {
    it('should match to snapshot - Primary', () => {
      const component = shallow(<SearchHeader />);
      expect(toJson(component)).toMatchSnapshot(
        'Primary SearchHeader  snapshot',
      );
    });
  });
});
