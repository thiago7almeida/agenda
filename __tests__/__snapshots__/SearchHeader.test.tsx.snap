// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHeader Rendering should match to snapshot - Primary: Primary SearchHeader  snapshot 1`] = `
<Layout
  level="1"
  style={
    Object {
      "paddingBottom": 8,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <Input
    accessoryRight={[Function]}
    onChangeText={[Function]}
    placeholder="Buscar"
    value=""
  />
</Layout>
`;
